docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,420 @@
+    },
+    continuationToken: null,
+    Mock.Of<Response>());
+```
+
+## [C#](#tab/csharp)
+
+```csharp
+Page responsePage = Page.FromValues(
+    new[]
+    {
+        new SecretProperties("secret1"),
+        new SecretProperties("secret2")
+    }, 
+    continuationToken: null,
+    new TestResponse());
+```
+
+---
+
+<xref:Azure.AsyncPageable%601> and <xref:Azure.Pageable%601> are classes that represent collections of models returned by the service in pages. The only difference between them is that one is used with synchronous methods while the other is used with asynchronous methods.
+
+To create a test instance of `Pageable` or `AsyncPageable`, use the `FromPages` static method:
+
+```csharp
+Page page1 = Page.FromValues(
+    new[]
+    {
+        new SecretProperties("secret1"),
+        new SecretProperties("secret2")
+    },
+    "continuationToken",
+    Mock.Of<Response>());
+
+Page page2 = Page.FromValues(
+    new[]
+    {
+        new SecretProperties("secret3"),
+        new SecretProperties("secret4")
+    },
+    "continuationToken2",
+    Mock.Of<Response>());
+
+Page lastPage = Page.FromValues(
+    new[]
+    {
+        new SecretProperties("secret5"),
+        new SecretProperties("secret6")
+    },
+    continuationToken: null,
+    Mock.Of<Response>());
+
+Pageable pageable = Pageable.FromPages(new[] { page1, page2, lastPage });
+AsyncPageable asyncPageable = AsyncPageable.FromPages(new[] { page1, page2, lastPage });
+```
+
+## Write a mocked unit test
+
+Suppose your app contains a class that finds the names of keys that will expire within a given amount of time.
+
+```csharp
+using Azure.Security.KeyVault.Secrets;
+
+public class AboutToExpireSecretFinder
+{
+    private readonly TimeSpan _threshold;
+    private readonly SecretClient _client;
+
+    public AboutToExpireSecretFinder(TimeSpan threshold, SecretClient client)
+    {
+        _threshold = threshold;
+        _client = client;
+    }
+
+    public async Task<string[]> GetAboutToExpireSecrets()
+    {
+        List<string> secretsAboutToExpire = new();
+
+        await foreach (var secret in _client.GetPropertiesOfSecretsAsync())
+        {
+            if (secret.ExpiresOn.HasValue &&
+                secret.ExpiresOn.Value - DateTimeOffset.Now <= _threshold)
+            {
+                secretsAboutToExpire.Add(secret.Name);
+            }
+        }
+
+        return secretsAboutToExpire.ToArray();
+    }
+}
+```
+
+You want to test the following behaviors of the `AboutToExpireSecretFinder` to ensure they continue working as expected:
+
+* Secrets that don't have an expiry date set are not returned.
+* Secrets with an expiry date closer to the current date than the threshold are returned.
+
+When unit testing you only want the unit tests to verify the application logic and not whether the Azure service or SDK works correctly. The following example tests the key behaviors using the popular xUnit framework for C#:
```

Comment:
Use "library" instead of "SDK" in this case, since we're referring to a single .NET library, not a collection of libraries.
```suggestion
When unit testing you only want the unit tests to verify the application logic and not whether the Azure service or library works correctly. The following example tests the key behaviors using the popular xUnit framework for C#:
```